virttest/env_process.py

```diff
@@ -314,6 +314,15 @@ def check_image(test, params, image_name, vm_process_status=None):
                               " Skip the image check.")
                 check_image_flag = False
 
+    # Save the potential bad image when the test is not passed.
+    # It should before image check.
+    if params.get("save_image", "no") == "yes":
+        if vm_process_status == "dead":
+            name = ("JOB-%s-TEST-%s-%s" % (
+                test.job.unique_id[:7], str(test.name.uid),
+                os.path.basename(image.image_filename)))
```

Comment:
Good idea. One issue is that multiple images of the same base-name would override this image. How about using `image_name` instead of the `image_filename`? Yes, the cons is it'd be harder to get the source image name, but that can be obtained from test params, on the other hand `image_name` should be unique (and if not then they use the same file).

 I'm wondering whether this name might not be "too-long", how about using `avocado.utils.astring.string_to_safe_path` to avoid potential failures?

Reply:
Note if you want to keep using the `image_filename` then you could add short hash of `dirname` of the `image_filename` followed by `basename` of the `image_filename`, therefor:

```
/foo/bar/test.qcow2 => 2340-test.qcow2
/foo/bar/test2.qcow2 => 2340-test2.qcow2
/baz/bar/test.qcow2 => 07bc-test.qcow2
```

Reply:
just for the record, with image_name the situation would be:

```
/foo/bar/test.qcow2 => $TEST_PREFIX-image1
/foo/bar/test2.qcow2 => $TEST_PREFIX-stg1
/baz/bar/test.qcow2 => $TEST_PREFIX-stg2
```

Reply:
It only uses the actual filename without path, as in https://github.com/avocado-framework/avocado-vt/pull/1868#issuecomment-458889283:
/home/kvm_autotest_root/images/`test.raw` -> /home/kvm_autotest_root/images/`JOB-11e4424-TEST-1-test.raw` 
Very image file will be copied to where it comes from, as shown above.